java-manual/modules/ROOT/pages/query-simple.adoc

```diff
@@ -0,0 +1,363 @@
+====
+
+
+== Delete from the database
+To remove a node and any relationship attached to it, use the Cypher clause link:{neo4j-docs-base-uri}/cypher-manual/current/clauses/delete/[`DETACH DELETE`]:
+
+.Remove the `Alice` node
+[source, java]
+----
+// import org.neo4j.driver.QueryConfig;
+
+// This does not delete _only_ p, but also all its relationships!
+var result = driver.executableQuery("""
+    MATCH (p:Person {name: $name})
+    DETACH DELETE p
+    """)
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .withParameters(Map.of("name", "Alice"))
+    .execute();
+var summary = result.summary();
+System.out.println("Query updated the database?");
+System.out.println(summary.counters().containsUpdates());
+----
+
+
+[#query-parameters]
+== Query parameters
+
+*Do not hardcode or concatenate parameters directly into queries*.
+Instead, always use placeholders and specify the link:{neo4j-docs-base-uri}/cypher-manual/current/syntax/parameters/[Cypher parameters], as shown in the previous examples.
+This is for:
+
+1. *performance benefits*: Neo4j compiles and caches queries, but can only do so if the query structure is unchanged;
+2. *security reasons*: see link:https://neo4j.com/developer/kb/protecting-against-cypher-injection/[protecting against Cypher injection].
+
+You may provide query parameters as a map through the `.withParameters()` method.
+
+[source, java]
+----
+var result = driver.executableQuery("MATCH (p:Person {name: $name})")
+    .withParameters(Map.of("name", "Alice"))  // mark-line
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .execute();
+----
+
+[NOTE]
+There can be circumstances where your query structure prevents the usage of parameters in all its parts.
+For those advanced use cases, see xref:query-advanced#_dynamic_values_in_property_keys_relationship_types_and_labels[Dynamic values in property keys, relationship types, and labels].
+
+
+== Error handling
+
+Because `.executableQuery()` can potentially raise a number of different link:https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/exceptions/package-summary.html[exceptions], the best way to handle errors is to catch all exceptions in a single `try/catch` block:
+
+[source, java]
+----
+try {
+    driver.executableQuery(...).execute()
+} catch (Exception as e) {
+    ...  // handle exception
+}
+----
+
+[TIP]
+The driver automatically retries to run a failed query, if the failure is deemed to be transient (for example due to temporary server unavailability).
+
+
+== Query configuration
+
+You can supply further configuration parameters to alter the default behavior of `.executableQuery()`.
+You do so through the method `.withConfig()`, which takes a link:https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/QueryConfig.html[`QueryConfig`] object.
+
+
+=== Database selection
```

Comment:
Just a general remark: I find this absolutely terrible and am wishing for quite some time that if no database is set on the driver, the drivers just should try `neo4j` and be done with it.

The examples look horrible with that dance and feel very much "in your face".

Reply:
Agreed, but what can we do. 

They also look horrible because the query config methods are defined on `QueryConfig` rather than directly on `ExecutableQuery`. If it looked like this, it wouldn't be as bad:
```java
var result = driver.executableQuery("MATCH (p:Person {name: $name})")
    .withParameters(Map.of("name", "Alice"))
    .withDatabase("neo4j")
    .execute();
```
I planned to bring this up somewhere else. All this config ballet (also in sessions and drivers) looks like madness to me.